src/app/plugins/scores.plugin.ts

```diff
@@ -29,98 +29,99 @@ export class ScoresPlugin extends Plugin {
-
-            visitorData = visitorData[0];
-            homeData = homeData[0];
-
-            console.log(visitorData);
-            console.log(homeData);
-
-            visitorData = visitorData.split('%20');
-            homeData = homeData.split('%20');
-
-            visitorData[0] = visitorData[0].slice(1); //Remove the prefix that starts the visitors name
-
-            const visitorName = this._getTeamNameFromTeamData(visitorData);
-            const homeName = this._getTeamNameFromTeamData(homeData);
-
-            const visitorScore = parseInt(visitorData[visitorData.length - 1]);
-            const homeScore = parseInt(homeData[homeData.length - 1]);
-
-            //Boolean for if the searched team is the one winning
-            let winning = 0;
-            let opponentName = '';
-            let teamName = '';
-
-            //If searched team is the visitor, else
-            if (this._containsAllTokens(visitorName, desiredTeamQuery.split(' '))) {
-
-              if (visitorScore > homeScore) { winning = 1; }  //Winning
-              else if (visitorScore < homeScore) { winning = 0; } //Losing
-              else if (visitorScore == homeScore) { winning = 2; } //Tied
-
-              opponentName = homeName;
-              teamName = visitorName;
-            } else {
-
-              if (homeScore > visitorScore) { winning = 1; }  //Winning
-              else if (homeScore < visitorScore) { winning = 0; } //Losing
-              else if (homeScore == visitorScore) { winning = 2; } //Tied
-
-              opponentName = visitorName;
-              teamName = homeName;
-            }
-
-            let winningString;
-            switch (winning) {
-              case 0: winningString = 'losing';
-                break;
-              case 1: winningString = 'winning';
-                break;
-              case 2: winningString = 'tied';
-                break;
-              default:
-                winningString = 'unknown';
-                break;
-            }
-
-            const embed = new RichEmbed();
-            embed.setTitle(visitorName + ' at ' + homeName);
-            embed.setColor('#7289da');
-            embed.setDescription(
-              teamName + ' is currently ' + winningString + ' against ' + opponentName
-            );
-            embed.setFooter(visitorScore + ' - ' + homeScore);
-            message.channel.send(embed);
-            messagesSent++;
+
+        if (!this._containsAllTokens(game, desiredTeamQuery.split(' '))) { continue; } //Check if game does not contain all search terms
+        if (messagesQueued >= this._MAX_NUM_DISPLAY) { break; } //Stop if the max amount of messages have been queued
+
+        teamFound = true; //Designates as true if any game was found with matching search term
+
+        let visitorData = game.match(visitorDataRegex);
+        let homeData = game.match(homeDataRegex);
+
+        //Makes sure regex found data
+        if (visitorData && homeData) {
```

Comment:
Same pattern here. Instead of having the entire rest of the loop in this if block, just do the pre-emptive negative check like you did earlier.